ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2

```diff
@@ -129,6 +129,45 @@
 | {{ policy.name }} | > {{ min_digits }} | > {{ min_length }} | > {{ min_lower }} | > {{ min_special }} | > {{ min_upper }} | < {{ max_repetitive }} | < {{ max_sequential }} |
 {%         endfor %}
 {%     endif %}
+{%     if management_security.shared_secret_profiles is arista.avd.defined %}
+
+### Secret Profiles
+
+| Profile Name | Secret Name | Secret Password | Receive Lifetime | Transmit Lifetime |
+| ------------ | ----------- | --------------- | ---------------- | ----------------- |
+{%         for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile_name") %}
+{%             if profile.secrets is arista.avd.defined %}
+{%                 set temp = [] %}
+{%                 for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%                     do temp.append(secret) %}
+{%                 endfor %}
```

Comment:
Please avoid creating another list. The code here is really hard to follow, and it is not clear to me why you have chosen a single table if you really want a table per profile. (see other documentation sections for examples, where we create a table per entity.